### PR TITLE
Optimise alignment/part2 avx2 optimisations

### PR DESCRIPTION
**Pull request by rrahn**
Adds avx2 intrinsic implementations for transpose and extract. 
This makes the transformation of the sequence batches to simd vectors faster.

Part of #1804

### CODE REVIEW

File: include/seqan3/core/simd/detail/simd_algorithm_avx2.hpp [hunk 1]
```diff
@@ -83,7 +83,59 @@ constexpr simd_t load_avx2(void const * mem_addr)
...
+    };
+
+    auto _mm256_unpackhi_epi128 = [] (__m256i const & a, __m256i const & b)
+    {
+        return _mm256_permute2x128_si256(a, b, 0x31);
+    };
+
+    // we need a look-up table to reverse the lowest 4 bits
+    // in order to place the permute the transposed rows
+    static const uint8_t bitRev[] = { 0, 8, 4,12, 2,10, 6,14, 1, 9, 5,13, 3,11, 7,15,
```
[the remark this answers is not in the document]
Comment (rrahn) [reply include/seqan3/core/simd/detail/simd_algorithm_avx2.hpp:101]: ah good spot

File: include/seqan3/core/simd/simd_algorithm.hpp [hunk 2]
```diff
@@ -160,6 +160,20 @@ constexpr simd_t extract_halve(simd_t const & src)
     else // if constexpr (simd_traits<simd_t>::max_length == 16) // SSE4
         return detail::extract_halve_sse4<index>(src);
```
[the remark this answers is not in the document]
Comment (rrahn) [reply include/seqan3/core/simd/simd_algorithm.hpp:161]: yes, I also saw this

File: include/seqan3/core/simd/simd_algorithm.hpp [hunk 3]
```diff
@@ -373,6 +383,16 @@ constexpr void transpose(std::array<simd_t, simd_traits<simd_t>::length> & matri
 {
     detail::transpose_matrix_sse4(matrix);
 }
+// AVX2 implementation
+template <simd::simd_concept simd_t>
+    requires detail::is_builtin_simd_v<simd_t> &&
+             detail::is_native_builtin_simd_v<simd_t> &&
+             simd_traits<simd_t>::max_length == 32 &&
+             simd_traits<simd_t>::length == 32
```
Comment (marehr) [inline include/seqan3/core/simd/simd_algorithm.hpp:391]:
```suggestion
             (simd_traits<simd_t>::max_length == 32) &&
             (simd_traits<simd_t>::length == 32)
```

I saw the conflicts and we need parenthesis here

File: include/seqan3/core/simd/detail/simd_algorithm_avx2.hpp [hunk 4]
```diff
@@ -81,9 +81,59 @@ constexpr simd_t load_avx2(void const * mem_addr)
     return reinterpret_cast<simd_t>(_mm256_loadu_si256(reinterpret_cast<__m256i const *>(mem_addr)));
 }
 
-// TODO: not implemented and used yet, if you implement it don't forget to add it to seqan3::simd::transpose
```
Comment (smehringer) [inline include/seqan3/core/simd/detail/simd_algorithm_avx2.hpp:84]: I guess this was done or is not applicable anymore?

Comment (rrahn) [reply include/seqan3/core/simd/detail/simd_algorithm_avx2.hpp:84]: it is implemented now.

**Pull request merged**